pkg/constants/constants.go

```diff
@@ -19,10 +19,13 @@ package constants
 import "time"
 
 const (
-	// QueueAnnotation is the annotation in the workload that holds the queue name.
+	// QueueLabel is the label key in the workload that holds the queue name.
+	QueueLabel = "kueue.x-k8s.io/queue-name"
```

Comment:
Revisit this label, I'm reading the api designs about `MatchFields` https://github.com/kubernetes/kubernetes/blob/fc8b5668f506b6d71ddc5243a21f5d0c5f8271d3/pkg/apis/core/types.go#L2558-L2563

This field was designed specially for daemonSet pod scheduling, one reason why we avoid to reuse `MatchExpressions` is because we can't model label as non-label usages, in this case, it should be an unique localQueue name, not a set of queue names. Annotation is something we think not bad as an experimental feature, and considering we haven't populate `queueName` to job yet.

Reply:
Are you saying that we should stay with an annotation instead of a label?

I'm not understanding what you mean by a unique localQueue name. Multiple workloads can point to the same queue.

Reply:
I mean a workload can only belong to one localQueue, we're taking the name field into labels,, which seems like we're abusing the label.

Reply:
This label is for the (custom) Job objects, not the Workloads. We call them workloads in general. Maybe we should call them jobs? not sure.